lib/HLSL/HLLowerUDT.cpp

```diff
@@ -179,6 +179,9 @@ hlsl::TranslateInitForLoweredUDT(Constant *Init, Type *NewTy,
 }
 
 void hlsl::ReplaceUsesForLoweredUDT(Value *V, Value *NewV) {
+  // Merge GepUse first to avoid mutate type and merge gep use at same time.
+  dxilutil::MergeGepUse(V);
```

Comment:
I don't think it's ok to add this here.  This function is recursed into, and `V` may be referenced after the call, but MergeGepUse might replace `V` if it's a GEP when merging into another GEP.